packages/medusa/src/services/customer.ts

```diff
@@ -405,60 +378,88 @@ class CustomerService extends BaseService {
    * @param {Object} addrRepo - address repository
    * @return {Promise} the result of the update operation
    */
-  async updateBillingAddress_(customer, addressOrId, addrRepo) {
-    if (addressOrId === null) {
-      customer.billing_address_id = null
-      return
-    }
-
-    if (typeof addressOrId === `string`) {
-      addressOrId = await addrRepo.findOne({
-        where: { id: addressOrId },
-      })
-    }
+  async updateBillingAddress_(
+    customer: Customer,
+    addressOrId: string | DeepPartial<Address> | undefined
+  ): Promise<void> {
+    return await this.atomicPhase_(async (manager) => {
```

Comment:
todo: Remove atomicPhase, and update visibility to protected.
At the beginning of the method you can do that :)
```ts
const transactionManager = this.transactionManager_ ?? this.manager
```

Reply:
Why should we remove the atomicphase?

Reply:
Because it is a protected method, therefore it is not publicly available and can use the internal manager since it will be called by other methods that are in a transaction

Reply:
Yeah I agree, but this is something we discussed with seb and it seams that the convention was: atomic on public and internal transactionManager on private and protected methdos

Reply:
But i agree that the end result is the same, it is just a consistency comment 😃